core/tls.go

```diff
@@ -0,0 +1,80 @@
+package core
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"os"
+
+	"chain/errors"
+)
+
+var ErrNoTLS = errors.New("no TLS configuration available")
+
+// TLSConfig returns a TLS config suitable for use
+// as a Chain Core client and server.
+// It reads a PEM-encoded X.509 certificate and private key
+// from certFile and keyFile.
+// If rootCAs is given, it reads a list of trusted root CA certs
+// from the filesystem;
+// otherwise it uses the system cert pool.
+//
+// For compatibility, it attempts to read the cert and key
+// from the environment if certFile and keyFile both
+// do not exist in the filesystem.
+//   TLSCRT=[PEM-encoded X.509 certificate]
+//   TLSKEY=[PEM-encoded X.509 private key]
+//
+// If certFile and keyFile do not exist or are empty
+// and the environment vars are both unset,
+// TLSConfig returns ErrNoTLS.
+func TLSConfig(certFile, keyFile, rootCAs string) (*tls.Config, error) {
+	config := &tls.Config{
+		// This is the default set of protocols for package http.
+		// ListenAndServeTLS and Transport set this automatically,
+		// but since we're supplying our own TLS config,
+		// we have to set it here.
+		NextProtos: []string{"http/1.1", "h2"},
```

Comment:
just want to call out that currently cmd/cored disables http2:

```
// Disable HTTP/2 for now until the Go implementation is more stable.
// https://github.com/golang/go/issues/16450
// https://github.com/golang/go/issues/17071
TLSNextProto: map[string]func(*http.Server, *tls.Conn, http.Handler){},
```
If we do use this tls.Config on the http.Server it will start enabling it. That sounds fine to me but just wanted to call it out.